sklearn/decomposition/pca.py

```diff
@@ -175,25 +175,32 @@ class PCA(_BasePCA):
             0 < n_components < min(X.shape)
         randomized :
             run randomized SVD by the method of Halko et al.
+        lobpcg :
+            run Locally Optimal Block Preconditioned Conjugate Gradient
+            (LOBPCG) by Knyazev 2001 for a normal matrix X'*X or X*X',
+            whichever of the two is of the smallest size.
 
         .. versionadded:: 0.18.0
 
     tol : float >= 0, optional (default .0)
-        Tolerance for singular values computed by svd_solver == 'arpack'.
+        Tolerance for singular values computed by svd_solver == 'arpack' or
```

Comment:
It weird to keep 0. and replace by something else. 'auto' or None would be better.